File: JAVA/TeachingAssistant.java
// TeachingAssistant mewarisi Student (Multilevel Inheritance)
public class TeachingAssistant extends Student {
    private int jamAsistensi;
    private String mataKuliah;

    public TeachingAssistant() {
        super();
        this.jamAsistensi = 0;
        this.mataKuliah = "";
    }

    public TeachingAssistant(String id, String nama, int usia,
                             String nim, String jurusan, double gpa,
                             int jamAsistensi, String mataKuliah) {
        super(id, nama, usia, nim, jurusan, gpa);
        this.jamAsistensi = jamAsistensi;
        this.mataKuliah = mataKuliah;
    }

    // Getter & Setter
    public int getJamAsistensi() { return jamAsistensi; }
    public void setJamAsistensi(int jamAsistensi) { this.jamAsistensi = jamAsistensi; }
    public String getMataKuliah() { return mataKuliah; }
    public void setMataKuliah(String mataKuliah) { this.mataKuliah = mataKuliah; }
}

File: CPP/TeachingAssistant.cpp
#pragma once
#include "Student.cpp"

// TeachingAssistant mewarisi Student (Multilevel Inheritance)
class TeachingAssistant : public Student {
private:
    int jamAsistensi;
    string mataKuliah; // atribut baru

public:
    TeachingAssistant() : Student(), jamAsistensi(0), mataKuliah("") {}

    TeachingAssistant(string id, string nama, int usia, 
                      string nim, string jurusan, double gpa,
                      int jamAsistensi, string mataKuliah)
        : Student(id, nama, usia, nim, jurusan, gpa),
          jamAsistensi(jamAsistensi), mataKuliah(mataKuliah) {}

    // Getter
    int getJamAsistensi() const { return jamAsistensi; }
    void setJamAsistensi(int jam) { jamAsistensi = jam; }

    // Setter
    string getMataKuliah() const { return mataKuliah; }
    void setMataKuliah(string mk) { mataKuliah = mk; }
};